lib/rspec/mocks/proxy.rb

```diff
@@ -266,6 +266,10 @@ def find_matching_method_stub(method_name, *args)
       def find_almost_matching_stub(method_name, *args)
         method_double_for(method_name).stubs.find { |stub| stub.matches_name_but_not_args(method_name, *args) }
       end
+
+      def with_range(expectation)
+        expectation.expectation_count_type || !expectation.ordered? ? (0..-1) : @order_group.message_range
```

Comment:
I feel this would read better as guard clauses rather than a compound ternary...

```Ruby
return expectation.expectation_count_type if expectation.expectation_count_type
return @order_group.message_range if expectation.ordered?
0..-1
```